src/Elm/Generic.hs

```diff
@@ -118,32 +119,68 @@ instance Elm Double where toElmDefinition _ = DefPrim ElmFloat
 instance Elm Text    where toElmDefinition _ = DefPrim ElmString
 instance Elm LT.Text where toElmDefinition _ = DefPrim ElmString
 
-instance Elm Value where toElmDefinition _ = DefPrim ElmValue
+instance (Elm a, Elm b) => Elm (a, b) where
+    toElmDefinition _ = DefPrim $ ElmPair (elmRef @a) (elmRef @b)
+
+instance (Elm a, Elm b, Elm c) => Elm (a, b, c) where
+    toElmDefinition _ = DefPrim $ ElmTriple (elmRef @a) (elmRef @b) (elmRef @c)
 
 -- TODO: should it be 'Bytes' from @bytes@ package?
 -- https://package.elm-lang.org/packages/elm/bytes/latest/Bytes
 -- instance Elm B.ByteString  where toElmDefinition _ = DefPrim ElmString
 -- instance Elm LB.ByteString where toElmDefinition _ = DefPrim ElmString
 
-instance Elm UTCTime where toElmDefinition _ = DefPrim ElmTime
+----------------------------------------------------------------------------
+-- Builtin instances
+----------------------------------------------------------------------------
+
+instance Elm Value where
+    toElmDefinition _ = DefBuiltin $ ElmBuiltin
+        { builtinImplType = "Value"
+        , builtinImplEncoder = "Basics.identity"
+        , builtinImplDecoder = "D.value"
+        , builtinImplParams = []
+        }
+
+instance Elm UTCTime where
+    toElmDefinition _ = DefBuiltin $ ElmBuiltin
+        { builtinImplType = "Posix"
+        , builtinImplEncoder = "Iso.encode"
+        , builtinImplDecoder = "Iso.decoder"
+        , builtinImplParams = []
+        }
 
 instance Elm a => Elm (Maybe a) where
-    toElmDefinition _ = DefPrim $ ElmMaybe $ elmRef @a
+    toElmDefinition _ = DefBuiltin $ ElmBuiltin
+        { builtinImplType = "Maybe"
+        , builtinImplEncoder = "elmStreetEncodeMaybe"
+        , builtinImplDecoder = "nullable"
+        , builtinImplParams = [elmRef @a]
+        }
 
 instance (Elm a, Elm b) => Elm (Either a b) where
-    toElmDefinition _ = DefPrim $ ElmResult (elmRef @a) (elmRef @b)
-
-instance (Elm a, Elm b) => Elm (a, b) where
-    toElmDefinition _ = DefPrim $ ElmPair (elmRef @a) (elmRef @b)
-
-instance (Elm a, Elm b, Elm c) => Elm (a, b, c) where
-    toElmDefinition _ = DefPrim $ ElmTriple (elmRef @a) (elmRef @b) (elmRef @c)
+    toElmDefinition _ = DefBuiltin $ ElmBuiltin
+        { builtinImplType = "Result"
+        , builtinImplEncoder = "elmStreetEncodeEither"
+        , builtinImplDecoder = "elmStreetDecodeEither"
+        , builtinImplParams = [elmRef @a, elmRef @b]
+        }
 
 instance Elm a => Elm [a] where
-    toElmDefinition _ = DefPrim $ ElmList (elmRef @a)
+    toElmDefinition _ = DefBuiltin $ ElmBuiltin
+        { builtinImplType = "List"
+        , builtinImplEncoder = "E.list"
```

Comment:
The E. and D. in this definition seem magical. Where are they coming from?

Reply:
yes. I agreee this is not nice though. But I would probably rather address that in separate PR